node/pkg/processor/processor.go

```diff
@@ -82,6 +83,14 @@ type (
 	}
 )
 
+func (s *state) MessageID() string {
+	if s.ourObservation != nil {
+		return s.ourObservation.MessageID()
+	}
+
+	return hex.EncodeToString(s.txHash)
```

Comment:
Is this safe to sustitute, generally speaking? Should there be a comment on this method that this should not be used to uniquely identify `state` structs?

Reply:
Changed it to `LoggingID` and added a comment.